src/Zynga/Framework/IO/Disk/V1/Manager.hh

```diff
@@ -0,0 +1,243 @@
+<?hh // strict
+
+namespace Zynga\Framework\IO\Disk\V1;
+
+use \Exception;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToCloseFileException;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToCreateDirectoryException;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToOpenFileException;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToWriteToFileException;
+use Zynga\Framework\IO\Disk\V1\Exception\ReadPermissionsException;
+use Zynga\Framework\IO\Disk\V1\Exception\WritePermissionsException;
+use Zynga\Framework\IO\Disk\V1\ManagerInterface as DiskIOManagerInterface;
+
+/**
+ * Lightweight class for managing Disk IO
+ */
+class Manager implements DiskIOManagerInterface {
+
+  private static ?DiskIOManagerInterface $instance;
+
+  public static function instance(): DiskIOManagerInterface {
+    if (self::$instance === null) {
+      self::$instance = new self();
+    }
+    invariant(self::$instance !== null, '$instance was null');
+
+    return self::$instance;
+  }
+
+  protected function __construct() {
+  }
+
+  /**
+   * See @ManagerInterface
```

Comment:
Just an FYI:
Javadoc style links to other class names and interfaces use the '@see' tag